src/redisearch_rs/inverted_index/src/lib.rs

```diff
@@ -38,6 +39,15 @@ impl From<Delta> for usize {
     }
 }
 
+// Non-numeric encoders only accept deltas that fit in a `u32`.
+impl TryFrom<Delta> for u32 {
+    type Error = std::num::TryFromIntError;
+
+    fn try_from(delta: Delta) -> Result<Self, Self::Error> {
+        u32::try_from(delta.0)
+    }
+}
```

Comment:
Shouldn't this be covered by a test?

Reply:
Actually the coverage is right. This test checks [this code path](https://github.com/RediSearch/RediSearch/blob/rs_ii_enc_dec/src/redisearch_rs/inverted_index/src/freqs_only.rs#L32) not this code.

I double checked and we don't need this `TryFrom`. We may need something like this if we enforce the two Delta sizes in the API but we agreed to defer that for now (see https://github.com/RediSearch/RediSearch/pull/6378#discussion_r2174848606 ).

So I just removed this code for now.